zuul-core/src/main/java/com/netflix/zuul/netty/timeouts/OriginTimeoutManager.java

```diff
@@ -78,25 +58,20 @@ public IClientConfig getRequestClientConfig(HttpRequestMessage zuulRequest) {
         return overriddenClientConfig;
     }
 
-    public int computeReadTimeout(IClientConfig requestConfig, int attempt) {
-        Duration readTimeout = getReadTimeout(requestConfig, attempt);
-        return Math.toIntExact(readTimeout.toMillis());
-    }
-
     /**
      * Derives the read timeout from the configuration.  This implementation prefers the longer of either the origin
      * timeout or the request timeout.
      *
-     * @param requestConfig the config for the request.
-     * @param attemptNum    the attempt number, starting at 1.
+     * @param request the request.
+     * @param attemptNum the attempt number, starting at 1.
      */
-    protected Duration getReadTimeout(IClientConfig requestConfig, int attemptNum) {
+    public Duration computeReadTimeout(HttpRequestMessage request, int attemptNum) {
         Long noTimeout = null;
         // TODO(carl-mastrangelo): getProperty is deprecated, and suggests using the typed overload `get`.   However,
         //  the value is parsed using parseReadTimeoutMs, which supports String, implying not all timeouts are Integer.
         //  Figure out where a string ReadTimeout is coming from and replace it.
         Long originTimeout = parseReadTimeoutMs(getOriginReadTimeout(noTimeout));
-        Long requestTimeout = parseReadTimeoutMs(getRequestReadTimeout(requestConfig, noTimeout));
+        Long requestTimeout = parseReadTimeoutMs(getRequestReadTimeout(request, noTimeout));
```

Comment:
I feel this is a code smell.  The `request` is being passed into `getRequestReadTimeout`, but only the config is being used.   Rather than `getRequestReadTimeout` being responsible for extracting the config, it seems like it should be passed in, which would mean extracting it just above this line, and passing the config in.